src/test_typing_extensions.py

```diff
@@ -7312,6 +7312,62 @@ def test_no_instance_subclassing(self):
             class MyAlias(TypeAliasType):
                 pass
 
+    def test_type_params_compatibility(self):
+        # Regression test to assure compatibility with typing variants
+        with self.subTest(type_params="typing.TypeVar"):
+            TypeAliasType("TypingTypeParams", ..., type_params=(typing.TypeVar('T'),))
+        with self.subTest(type_params="typing.TypeAliasType"):
+            if not hasattr(typing, "TypeAliasType"):
+                self.skipTest("typing.TypeAliasType is not available before 3.12")
+            TypeAliasType("TypingTypeParams", ..., type_params=(typing.TypeVarTuple("Ts"),))
+        with self.subTest(type_params="typing.TypeAliasType"):
```

Comment:
This is the same `type_params` as before. As on some other PRs, seems better to use separate test methods.